Editor/Scenes/SongEditor/SongEditor.gd

```diff
@@ -34,15 +45,46 @@ func add_track(instrument: Button):
 	var inst := GoDAW.get_instrument(instrument.text)
 	$Sequencer.INSTRUMENTS[instrument.text] = inst
 
-func sequence():
+func check_error(script_location: String) -> String:
+	# Error check
+	var err = []
+	var _n = OS.execute(BIN, ["-s", script_location, "--check-only", "--no-window"], true, err, true)
+	var error = (err[0] as String).substr(err[0].find("SCRIPT"))\
+		.replace(script_location, "SongScript")\
+		.replace(" (", "")\
+		.replace("SCRIPT ERROR: ", "")\
+		.replace("GDScript::reload", "")
+	var prev = 0
+	for i in error.count("SongScript:"):
+		prev = error.find(")", error.find("SongScript"))
+		error[prev] = ""
+	return error
```

Comment:
Is there a more machine-friendly output for this? Or is it possible to use the godot binary as a language server? If not, can we use regex instead to extract these parts?

Reply:
There is indeed an LSP but not sure about if the executable for the application would have that... But even if there is a LSP handling all cases....